gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,241 @@
+        return cls._from_dict(info)
+
+    @classmethod
+    def _list(cls, client, filter=None):
+        """List all metric descriptors for the project.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type filter: string or None
+        :param filter: An optional filter string describing the metric
+            descriptors to be returned. See the `filter documentation`_.
+
+        :rtype: list of :class:`MetricDescriptor`
+        :returns: A list of metric descriptor instances.
+
+        .. _filter documentation:
+            https://cloud.google.com/monitoring/api/v3/filters
+        """
+        # Allow "filter" as a parameter name: pylint: disable=redefined-builtin
+
+        path = '/projects/{project}/metricDescriptors/'.format(
+            project=client.project)
+
+        descriptors = []
+
+        page_token = None
+        while True:
+            params = {}
+
+            if filter is not None:
+                params['filter'] = filter
+
+            if page_token is not None:
+                params['pageToken'] = page_token
+
+            response = client.connection.api_request(
+                method='GET', path=path, query_params=params)
+            for info in response.get('metricDescriptors', []):
+                descriptors.append(cls._from_dict(info))
+
+            page_token = response.get('nextPageToken')
+            if not page_token:
+                break
+
+        return descriptors
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a metric descriptor from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`MetricDescriptor`
+        :returns: A metric descriptor.
+        """
+        return cls(
+            name=info['name'],
+            type=info['type'],
+            labels=tuple(LabelDescriptor._from_dict(label)
+                         for label in info.get('labels', [])),
+            metric_kind=info['metricKind'],
+            value_type=info['valueType'],
+            unit=info.get('unit', ''),
+            description=info.get('description', ''),
+            display_name=info.get('displayName', ''),
+        )
+
+    def __repr__(self):
+        return (
+            'MetricDescriptor(\n'
+            ' name={name!r},\n'
+            ' type={type!r},\n'
+            ' metric_kind={metric_kind!r}, value_type={value_type!r},\n'
+            ' labels={labels!r},\n'
+            ' display_name={display_name!r}, unit={unit!r},\n'
+            ' description={description!r})'
+        ).format(**self.__dict__)
+
+
+class Metric(collections.namedtuple('Metric', 'type labels')):
+    """A specific metric identified by specifying values for all labels.
+
+    This is a named tuple. See :func:`collections.namedtuple`.
```

Comment:
This comment is unnecessary. For those that care, they can just use `type(metric).__bases__` and for those that don't they'll be happy (happily ignorant?) they can use `metric[0]` and `metric.type` interchangeably.